kernel/private/rest/classes/ezpkernelrest.php

```diff
@@ -0,0 +1,294 @@
+     */
+    protected function sessionInit()
+    {
+    }
+
+    /**
+     * Execution point for controller actions.
+     * Returns false if not supported
+     *
+     * @return ezpKernelResult
+     */
+    public function run()
+    {
+        $mvcConfig = new ezpMvcConfiguration( $this->responseWriterClass );
+        $frontController = new ezpMvcConfigurableDispatcher( $mvcConfig );
+        return $frontController->run();
+    }
+
+    /**
+     * Not supported by ezpKernelTreeMenu
+     *
+     * @throws \RuntimeException
+     */
+    public function runCallback( \Closure $callback, $postReinitialize = true )
+    {
+        throw new \RuntimeException( 'runCallback() method is not supported by ezpKernelTreeMenu' );
+    }
+
+    /**
+     * Sets whether to use exceptions inside the kernel.
+     *
+     * @param bool $useExceptions
+     */
+    public function setUseExceptions( $useExceptions )
+    {
+        eZModule::$useExceptions = (bool)$useExceptions;
+    }
+
+    /**
+     * Reinitializes the kernel environment.
+     *
+     * @return void
+     */
+    public function reInitialize()
+    {
+        $this->isInitialized = false;
+    }
+
+    /**
+     * Runs the shutdown process
+     */
+    protected function shutdown( $reInitialize = true )
+    {
+        eZExecution::cleanup();
+        eZExecution::setCleanExit();
+        eZExpiryHandler::shutdown();
+        if ( $reInitialize )
+            $this->isInitialized = false;
+    }
+
+    /**
+     * Handles an internal error.
+     * If not using exceptions, will return an ezpKernelResult object with JSON encoded message.
+     *
+     * @param string $errorMessage
+     * @param int $errorCode
+     * @return ezpKernelResult
+     *
+     * @throws RuntimeException
+     * @throws ezpAccessDenied
+     *
+     * @see setUseExceptions()
+     */
+    protected function exitWithInternalError( $errorMessage, $errorCode = 500 )
+    {
+        if ( eZModule::$useExceptions )
+        {
+            switch ( $errorCode )
+            {
+                case 403:
+                    throw new ezpAccessDenied( $errorMessage );
+
+                case 500:
+                default:
+                    throw new RuntimeException( $errorMessage, $errorCode );
+            }
+        }
+        else
+        {
+            header( $_SERVER['SERVER_PROTOCOL'] . ' 500 Internal Server Error' );
+            eZExecution::cleanup();
+            eZExecution::setCleanExit();
+
+            return new ezpKernelResult(
+                json_encode(
+                    array(
+                        'error'        => $errorMessage,
+                        'code'         => $errorCode
+                    )
+                )
+            );
+        }
+    }
+
+    /**
+     * Checks whether the kernel handler has the Symfony service container
+     * container or not.
+     *
+     * @return bool
+     */
+    public function hasServiceContainer()
+    {
+        return false;
```

Comment:
For us: nothing. But can be useful using custom code... The same goes for `ezpKernelWeb` :wink: